### DISCUSSION TITLE
Welcome to RDKit Discussions!

### DISCUSSION THREAD

**Discussion by greglandrum** [rdkit/rdkit#3560] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect the RDKit community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with the rest of the community.
  * Be welcoming, friendly, and open-minded. The RDKit has a great community and that should be reflected here.

**Comment by amorgat** [comment]
Hi RDKit,

I am calculating the similarity between enzymatic reactions (www.rhea-db.org) using CreateDifferenceFingerprintForReaction with the different fingerprint types:
- AtomPairFP 
- MorganFP
- TopologicalTorsion

In ReactionFingerprintParams, we can define fpSize and fpType, but how do we define the parameters for the fp? 
For example, what radius is used for MorganFP, is it possible to change the default value? 
What are the default values used for AtomPairFP, MorganFP, and TopologicalTorsion?

Best, Anne

**Reply by liudongliangHI** [reply]
Hello, amorgat. Coincidentally, I'm doing the same thing. I found an earlier discussion (see the [link](https://sourceforge.net/p/rdkit/mailman/message/35252296/) ) , and I modified it on top of that. Although some time has passed, perhaps you have solved this problem. But anyway, I hope the following code is useful to you. Please let me know if you have any comments on this.

```python
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit import DataStructs
import copy

def _createFP(mol,maxPath,fpType='RDKitFP'):
    mol.UpdatePropertyCache(False)
    if fpType == 'AtomPair':
        return AllChem.GetAtomPairFingerprint(mol, minLength=1,maxLength=maxPath)
    elif fpType == 'TopologicalTorsion':
        return AllChem.GetTopologicalTorsionFingerprint(mol)
    elif fpType == 'Morgan':
        Chem.GetSSSR(mol)
        return AllChem.GetMorganFingerprint(mol, radius=maxPath)
    else:
        Chem.GetSSSR(mol)
        return AllChem.GetMorganFingerprint(mol, radius=maxPath)

def getSumFps(fps):
    summedFP = copy.deepcopy(fps[0])
    for fp in fps[1:]:
        summedFP += fp
    return summedFP

def buildReactionFP(rxn, maxPath=1, fpType='Morgan'):
    reactants = rxn.GetReactants()
    products = rxn.GetProducts()
    rFP = getSumFps([_createFP(mol,maxPath,fpType=fpType) for mol in reactants])
    pFP = getSumFps([_createFP(mol,maxPath,fpType=fpType) for mol in products])
    return pFP-rFP

# construct the chemical reactions
rxn1 = AllChem.ReactionFromSmarts('[C:1]C1CCCCC1>>[N:1]C1CCCCC1')
rxn2 = AllChem.ReactionFromSmarts('[C:1]C1CCCNC1>>[N:1]C1CCCNC1')
rxn3 = AllChem.ReactionFromSmarts('[C:1]c1ccccc1>>[N:1]c1ccccc1')
```

I only entered reaction-product pairs here, but in fact, this function(AllChem.ReactionFromSmarts) also supports complete reactions.

```python
# construct difference fingerprint (subtracts reactant fingerprint from product)
fp1 = buildReactionFP(rxn1,maxPath=1,fpType='Morgan')
fp2 = buildReactionFP(rxn2,maxPath=1,fpType='Morgan')
fp3 = buildReactionFP(rxn3,maxPath=1,fpType='Morgan')

print(DataStructs.TanimotoSimilarity(fp1,fp2))
# 1.0
```
change radius for MorganFP, results will be different.
```python
fp1 = buildReactionFP(rxn1,maxPath=2,fpType='Morgan')
fp2 = buildReactionFP(rxn2,maxPath=2,fpType='Morgan')
fp3 = buildReactionFP(rxn3,maxPath=2,fpType='Morgan')

print(DataStructs.TanimotoSimilarity(fp1,fp2))
# 0.5
```
However, I don't know what its default molecular fingerprint parameters are, I didn't find it in the source code, I hope the official can be written in the documentation a little more clearly in the future. If you know, feel free to let me know.

**Reply by amorgat** [reply]
Thank you very much.
I guess the default parameters are those used in the publication (https://pubmed.ncbi.nlm.nih.gov/25541888/)
AtomPair: maxLength=3
Morgan: radius=2

**Reply by liudongliangHI** [reply]
Oh, I see, thanks.

**Comment by baoilleach** [comment]
Hi @greglandrum, is this forum an appropriate place to post a job advertisement relevant to RDKit users? Or better at rdkit-discuss (I see a few there)?

**Reply by greglandrum** [reply]
> Hi @greglandrum, is this forum an appropriate place to post a job advertisement relevant to RDKit users? Or better at rdkit-discuss (I see a few there)?

I think that RDKit-discuss and/or the LinkedIn group are better for job postings
